framework/src/base/FEProblemBase.C

```diff
@@ -2815,21 +2815,29 @@ FEProblemBase::execMultiAppTransfers(ExecFlagType type, MultiAppTransfer::DIRECT
   {
     const auto & transfers = wh.getActiveObjects();
 
+    #ifndef NDEBUG
     _console << COLOR_CYAN << "\nStarting Transfers on " <<  Moose::stringify(type) << string_direction << "MultiApps" << COLOR_DEFAULT << std::endl;
+    #endif
     for (const auto & transfer : transfers)
     {
       Moose::perf_log.push(transfer->name(), "Transfers");
       transfer->execute();
       Moose::perf_log.pop(transfer->name(), "Transfers");
     }
 
+    #ifndef NDEBUG
     _console << "Waiting For Transfers To Finish" << '\n';
     MooseUtils::parallelBarrierNotify(_communicator);
```

Comment:
Hmm, Skipping the barrier completely may have unintended consequences. I believe this was added because we wanted to make sure that data was available from all transfers before proceeding. In simple cases each processor just accesses local data so you won't see a problem but this may be a little aggressive to skip it altogether. 

At the very least we should just use a standard barrier to maintain the same functionality without printing.